cmd/minishift/cmd/root.go

```diff
@@ -89,6 +91,10 @@ var RootCmd = &cobra.Command{
 
 		constants.KubeConfigPath = filepath.Join(state.InstanceDirs.Machines, constants.MachineName+"_kubeconfig")
 
+		if minishiftStrings.Contains(cmdIgnoreCreatingMinishiftHomeDir, cmd.Name()) {
```

Comment:
Can we put the chec right at the top? or the initialization of the constants and InstancDir is necessary to run the commands?

Reply:
@anjannath , it seems it has be placed here only since if we place at the top then `cmd.Name()` which is returning `version` will affect `m openshift version` as a whole :)

Reply:
You could check for `cmd.Parent().Name()` and make a decision..